homeassistant/components/arcam_fmj/media_player.py

```diff
@@ -0,0 +1,355 @@
+"""Arcam media player."""
+import logging
+from typing import Optional
+
+from arcam.fmj import (
+    DecodeMode2CH,
+    DecodeModeMCH,
+    IncomingAudioFormat,
+    SourceCodes,
+)
+from arcam.fmj.state import State
+
+from homeassistant import config_entries
+from homeassistant.components.media_player import MediaPlayerDevice
+from homeassistant.components.media_player.const import (
+    MEDIA_TYPE_MUSIC,
+    SUPPORT_SELECT_SOUND_MODE,
+    SUPPORT_SELECT_SOURCE,
+    SUPPORT_TURN_ON,
+    SUPPORT_TURN_OFF,
+    SUPPORT_VOLUME_MUTE,
+    SUPPORT_VOLUME_SET,
+    SUPPORT_VOLUME_STEP,
+)
+from homeassistant.const import (
+    CONF_NAME,
+    CONF_ZONE,
+    CONF_HOST,
+    CONF_PORT,
+    SERVICE_TURN_ON,
+    STATE_OFF,
+    STATE_ON,
+)
+from homeassistant.helpers.typing import HomeAssistantType, ConfigType
+from homeassistant.helpers.service import async_call_from_config
+
+from .const import (
+    SIGNAL_CLIENT_DATA,
+    SIGNAL_CLIENT_STARTED,
+    SIGNAL_CLIENT_STOPPED,
+    DEFAULT_NAME,
+    DOMAIN_DATA_ENTRIES,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistantType,
+    config_entry: config_entries.ConfigEntry,
+    async_add_entities,
+):
+    """Set up the configuration entry."""
+    data = hass.data[DOMAIN_DATA_ENTRIES][config_entry.entry_id]
+    client = data["client"]
+    config = data["config"]
+
+    host = config[CONF_HOST]
+    port = config[CONF_PORT]
+
+    async_add_entities(
+        [
+            ArcamFmj(
+                State(client, zone),
+                zone_config.get(
+                    CONF_NAME,
+                    "{} ({}:{}) - {}".format(DEFAULT_NAME, host, port, zone),
```

Comment:
Can we move the default name to the config schema?

Reply:
Well i wanted it possible to override the full name not just prefix. So then I'd have to make it a format.

If I trusted the unique ID (now just host, port) since nothing else exist, I could drop config of name fully.

Reply:
Host, port  and zone are available in the config. Make a custom validator that sets the name if not already set.

Reply:
Might work. Its on parent level in config though. So validator need to access higher levels. Will ponder how to do that.

Reply:
Here's an example:
https://github.com/home-assistant/home-assistant/blob/c71a5643ffb0066fb81c122c6385b836db3c27c7/homeassistant/components/manual/alarm_control_panel.py#L46-L58

Reply:
Please do this.